libs/service-portal/settings/access-control/src/screens/Access/Access.tsx

```diff
@@ -209,153 +212,158 @@ function Access() {
     {} as GroupedApiScopes,
   )
 
+  const scopes = getValues()?.[SCOPE_PREFIX]?.map((item) => {
+    if (item.name && item.validTo)
+      return {
+        displayName: authApiScopes?.find((x) => x.name === item.name[0])
+          ?.displayName,
+        validTo: item.validTo,
+      }
+  })
+
   return (
     <Box>
       <IntroHeader
         title={authDelegation?.to?.name || ''}
         intro={defineMessage({
-          id: 'service.portal.settings.accessControl:access-intro',
+          id: 'sp.settings-access-control:access-intro',
           defaultMessage:
-            'Reyndu að lámarka þau réttindi sem þú vilt veita viðkomandi eins mikið og mögulegt er.',
+            'Hér velur þú hvaða aðgangur er veittur með þessu umboði og hversu lengi. Reyndu að lágmarka þau réttindi sem þú vilt veita viðkomandi eins mikið og mögulegt er.',
         })}
       />
       <FormProvider {...hookFormData}>
         <form onSubmit={onSubmit}>
-          <Box marginBottom={3} display="flex" justifyContent="flexEnd">
+          <Box marginBottom={8} display="flex" justifyContent="flexEnd">
             <Inline space={3}>
               {authDelegation?.scopes.length > 0 && (
-                <ModalBase
-                  baseId="authDelegation-remove"
-                  className={styles.modal}
-                  disclosure={
-                    <Button
-                      variant="ghost"
-                      colorScheme="destructive"
-                      size="small"
-                      icon="close"
-                    >
-                      {formatMessage({
+                <>
+                  <Button
+                    variant="text"
+                    colorScheme="destructive"
+                    size="small"
+                    icon="close"
+                    onClick={() => setCloseModalOpen(true)}
+                  >
+                    {formatMessage({
+                      id: 'sp.settings-access-control:access-remove-delegation',
+                      defaultMessage: 'Eyða umboði',
+                    })}
+                  </Button>
+                  {closeModalOpen && (
+                    <AccessModal
```

Comment:
Is this the accessibly correct way to open and close modals? Does it handle focus correctly? E.g. when going in and out of the modal?

Reply:
Oh lord - it was not. Thank you so much 🙏  I think that I fixed it ? 🤔